applications/MeshingApplication/custom_strategies/custom_convergencecriterias/error_mesh_criteria.h

```
@@ -297,16 +297,18 @@ class ErrorMeshCriteria : public virtual  ConvergenceCriteria< TSparseSpace, TDe
             // Remeshing
             if (mRemeshingUtilities == MMG)
             {
-                if (mDimension == 2)
-                {
-                    MmgProcess<2> MmgRemesh = MmgProcess<2>(mThisModelPart, mThisParameters["remeshing_parameters"]); 
-                    MmgRemesh.Execute();
-                }
-                else
-                {
-                    MmgProcess<3> MmgRemesh = MmgProcess<3>(mThisModelPart, mThisParameters["remeshing_parameters"]); 
-                    MmgRemesh.Execute();
-                }
+                #ifdef INCLUDE_MMG
+                    if (mDimension == 2)
+                    {
+                        MmgProcess<2> MmgRemesh = MmgProcess<2>(mThisModelPart, mThisParameters["remeshing_parameters"]); 
+                        MmgRemesh.Execute();
+                    }
+                    else
+                    {
+                        MmgProcess<3> MmgRemesh = MmgProcess<3>(mThisModelPart, mThisParameters["remeshing_parameters"]); 
+                        MmgRemesh.Execute();
+                    }
+                #endif
```

Would it make sense to throw an error if you get here but don't have MMG compiled? If I understand this correctly, this will just silently do nothing if one does not have MMG

---

I agree

---

It is the only way, the thing is that right now the only remesher I know is the MMG

---

Well, your code is perfectly fine, but I think an `#else KRATOS_ERROR << "Please compile with MMG to use this utility << std::endl;` or something of the sort would be helpful for end users.

---

Yes, indeed

---

Done